### PR TITLE
refactor: update `blas/ext/ssort2ins`  to follow current project conventions

### LINKED ISSUE #1535: [RFC]: refactor `blas/ext/base/ssort2ins` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/ssort2ins`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/ssort2ins) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by vr-varad**
## Description

> What is the purpose of this pull request?

This pull request refactors the ```@stdlib/blas/ext/ssort@ins``` to follow current project conventions
> ## Checklist
> * [x]  update readme.md
> * [x]  docs
> * [x]  src
> * [x]  include.gypi
> * [x]  manifest.json
> * [x]  package.json
> * [x]  lib
> * [x]  examples
> * [x]  benchmark
> * [x]  test

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1535

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by vr-varad** [comment]
@Pranavchiku I was not able to find any refactoring in benchmark, if there is any please let me know

**Comment by vr-varad** [comment]
Resolving package path...
Package path: /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/blas/ext/base/ssort2ins
Resolving package manifest...
Successfully resolved package manifest.
Resolving include directories...
Resolving source files...
Resolving libraries...
Resolving library paths...
Compiling example...
make[1]: Entering directory '/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/blas/ext/base/ssort2ins/examples/c'
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv/include/stdlib/napi/argv.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-float/include/stdlib/napi/argv_float.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-float/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/status-ok/include/stdlib/assert/napi/status_ok.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/is-type/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/is-type/include/stdlib/assert/napi/is_type.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/equal-types/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/equal-types/include/stdlib/assert/napi/equal_types.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv/include/stdlib/napi/argv.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-int64/include/stdlib/napi/argv_int64.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-int64/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/status-ok/include/stdlib/assert/napi/status_ok.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv/include/stdlib/napi/argv.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-strided-float64array/include/stdlib/napi/argv_strided_float64array.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-strided-float64array/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/status-ok/include/stdlib/assert/napi/status_ok.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv/include/stdlib/napi/argv.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-float64array/include/stdlib/napi/argv_float64array.h:22,
                 from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/napi/argv-float64array/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/status-ok/include/stdlib/assert/napi/status_ok.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/equal-typedarray-types/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/equal-typedarray-types/include/stdlib/assert/napi/equal_typedarray_types.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
In file included from /home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/is-typedarray/src/main.c:19:
/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/assert/napi/is-typedarray/include/stdlib/assert/napi/is_typedarray.h:22:10: fatal error: node_api.h: No such file or directory
   22 | #include <node_api.h>
      |          ^~~~~~~~~~~~
compilation terminated.
make[1]: *** [Makefile:124: example.out] Error 1
make[1]: Leaving directory '/home/varad/Desktop/stdlib/lib/node_modules/@stdlib/blas/ext/base/ssort2ins/examples/c'
Error when attempting to compile example.
ERROR: An error was encountered during execution.

make: *** [/home/varad/Desktop/stdlib/tools/make/lib/examples/c.mk:84: examples-c-files] Error 1

Encountered an error when running C examples

**Comment by vr-varad** [comment]
how could i resolve this error I think this is an error coming from addon.c @kgryte @Planeshifter

**Comment by kgryte** [comment]
@vr-varad See https://github.com/stdlib-js/stdlib/commit/07514b1f40412bc50348fe6a5c9a85c26f3c3675. You need to update your `manifest.json` configurations accordingly.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js [hunk 1]
```diff
@@ -58,18 +59,11 @@ function ssort2ins( N, order, x, strideX, offsetX, y, strideY, offsetY ) {
...
-	if ( strideY < 0 ) {
-		offsetY += (N-1) * strideY;
-	}
-	viewX = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offsetX), x.length-offsetX ); // eslint-disable-line max-len
-	viewY = new Float32Array( y.buffer, y.byteOffset+(y.BYTES_PER_ELEMENT*offsetY), y.length-offsetY ); // eslint-disable-line max-len
+	offsetX = minViewBufferIndex( N, strideX, offsetX );
+	offsetY = minViewBufferIndex( N, strideY, offsetY );
+
+	viewX = reinterpret( x, offsetX );
+	viewY = reinterpret( y, offsetY );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js]: This is incorrect and does not match the original logic. Namely, we need to handle the `order` value based on the sign of the input stride.

Furthermore, `reinterpret` is **not** the right API, as we are not looking to reinterpret a `Complex64Array` array. We need to create an offset view for `Float32Array`s.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json [hunk 2]
```diff
@@ -1,43 +1,47 @@
...
-			"libpath": [],
-			"dependencies": [
-				"@stdlib/math/base/assert/is-nanf",
-				"@stdlib/math/base/assert/is-negative-zerof"
-			]
-		}
-	]
+  "options": {},
+  "fields": [
+	{
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json]: This file has mixed tabs and spaces.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/src/addon.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/ssort2ins.h"
+#include "stdlib/blas/base/saxpy.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/src/addon.c]:
```suggestion
```

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/src/addon.c [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+*/
+static napi_value addon( napi_env env, napi_callback_info info ) {
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 6 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_FLOAT( env, alpha, argv, 1 );
+	STDLIB_NAPI_ARGV_INT64( env, strideX, argv, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, strideY, argv, 5 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, strideX, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, Y, N, strideY, argv, 4 );
+	c_ssort2ins( N, alpha, X, strideX, Y, strideY );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/src/addon.c]: You've simply copy and pasted from `saxpy`, but that is not what we want. We need to tailor argument extraction to `ssortins`.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/src/addon.c [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	return NULL;
+}
+
+STDLIB_NAPI_MODULE_EXPORT_FCN( addon )
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/src/addon.c]: Missing trailing newline. Ensure you have EditorConfig properly installed and setup locally, including integration with your local editor.

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this @vr-varad. In addition to the comments I left, this PR is missing changes to test files and other documentation examples, include `repl.txt` and JSDoc comments.

**Comment by vr-varad** [comment]
@kgryte @Planeshifter  I have made the necessary changes. Could you please review it?

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/examples/c/example.c [hunk 6]
```diff
@@ -40,4 +40,4 @@ int main( void ) {
 		printf( "x[ %i ] = %f\n", i, x[ i ] );
 		printf( "y[ %i ] = %"PRId64"\n", i, (int64_t)y[ i ] );
 	}
-}
+}
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/examples/c/example.c]: This change should not have been made.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js [hunk 7]
```diff
@@ -63,13 +64,14 @@ function ssort2ins( N, order, x, strideX, offsetX, y, strideY, offsetY ) {
 		flg *= -1.0; // reversing order
 		order *= -1.0;
 		strideX *= -1;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js:67]: I believe you need to remove this line. L69 is assuming the original `strideX`.

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js:67]: Hmm...this implementation is a bit trickier.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json [hunk 8]
```diff
@@ -1,43 +1,78 @@
...
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+        "dependencies": [
+          "@stdlib/napi/export",
+          "@stdlib/napi/argv",
+          "@stdlib/napi/argv-int64",
+          "@stdlib/napi/argv-strided-float64array"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json]: Should be `float32array` here.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json [hunk 9]
```diff
@@ -1,43 +1,78 @@
...
+      ],
+      "include": [
+        "./include"
+      ],
+      "libraries": [
+        "-lm"
+      ],
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/assert/is-nanf",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json]: Why are these deps here, but not in the other `dependencies` fields?

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js (hunk 7 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js:67]: I'd double check that the behavior you've introduced here matches the behavior of the previous implementation. In short, what is provided to `addon` to be the same before and after the changes introduced in this PR. And I'd confirm this for positive and negative strides.

**Comment by kgryte** [comment]
@vr-varad As mentioned above, this PR is still missing changes to test files and documentation.

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json (hunk 9 above)
Comment (vr-varad) [reply lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json]: @kgryte as they are only required in  examples and its use is nowhere else so I decided to put it here

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js (hunk 7 above)
Comment (vr-varad) [reply lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js:67]: So are there any changes required? @kgryte

Comment (kgryte) [reply lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js:67]: I'm not sure, which is why I encourage you to check. The implementation as is looks suspect, but I don't have the bandwidth atm to confirm.

**Comment by vr-varad** [comment]
@kgryte I tried to rectify everything as told but I am unable to find any changes to be made in test. And according to me the in ndarray.native.js both implementations achieve the same functionality of sorting two single-precision floating-point arrays based on the sort order of the first array using insertion sort. However, they differ slightly in how they handle input parameters and perform offset adjustments.

**Comment by kgryte** [comment]
You're right regarding tests. Just had a look through and those files seem fine, as is.

**Comment by kgryte** [comment]
`repl.txt` needs to be updated to remove the use of `floor`.

**Comment by vr-varad** [comment]
@kgryte in the repl.txt file, the floor function isn't being used. I removed it earlier. Could you please review it,

**Comment by vr-varad** [comment]
@kgryte are there any more changes to be made that can be resolved?
could I work on another issue?

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/README.md [hunk 10]
```diff
@@ -58,17 +58,14 @@ The function has the following parameters:
 -   **y**: second input [`Float32Array`][@stdlib/array/float32].
 -   **strideY**: `y` index increment.
 
-The `N` and `stride` parameters determine which elements in `x` and `y` are accessed at runtime. For example, to sort every other element
+The `N` and `stride` parameters determine which elements in the strided arrays are accessed at runtime. For example, to sort every other element
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/README.md:61]:
```suggestion
The `N` and stride parameters determine which elements in the strided arrays are accessed at runtime. For example, to sort every other element
```

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/README.md [hunk 11]
```diff
@@ -126,7 +121,7 @@ The function has the following additional parameters:
 -   **offsetX**: `x` starting index.
 -   **offsetY**: `y` starting index.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of `x`
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the offset parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of `x`.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/README.md:124]:
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the offset parameter supports indexing semantics based on a starting index. For example, to access only the last three elements of `x`
```

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/docs/repl.txt [hunk 12]
```diff
@@ -3,13 +3,14 @@
     Simultaneously sorts two single-precision floating-point strided arrays
     based on the sort order of the first array using insertion sort.
 
-    The `N` and `stride` parameters determine which elements in `x` and `y` are
-    accessed at runtime.
+    The `N` and stride parameters determine which elements in the strided 
+    arrays are are accessed at runtime.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/docs/repl.txt]:
```suggestion
    arrays are accessed at runtime.
```

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json [hunk 13]
```diff
@@ -1,43 +1,78 @@
...
+      "relative": false
+    },
+    {
+      "field": "libpath",
+      "resolve": true,
+      "relative": false
+    }
+  ],
+  "confs": [
+    {
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/manifest.json:28]: First configuration needs a `"task": "build",` property.

**Comment by vr-varad** [comment]
@Planeshifter @kgryte I'm encountering an error that's causing a bit of a roadblock for me. Would you be able to lend me a hand in troubleshooting it?

File: lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js [hunk 14]
```diff
@@ -58,18 +59,18 @@ function ssort2ins( N, order, x, strideX, offsetX, y, strideY, offsetY ) {
 	var viewY;
 	var flg;
 
-	flg = 1.0;
-	if ( strideX < 0 ) {
-		flg *= -1.0; // reversing order
+	flg = (strideX < 0) ? -1.0 : 1.0;
+	if (flg < 0) {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/ssort2ins/lib/ndarray.native.js]: This change doesn't add anything.

**Comment by kgryte** [comment]
@vr-varad It looks like the argument juggling in `lib/ndarray.native.js` is not correct; hence, the test failure.

**Comment by vr-varad** [comment]
@Planeshifter @kgryte I made the necessary changes and the issue is finally resolved Thank you @kgryte .
You can review it.

**Comment by vr-varad** [comment]
@Planeshifter @kgryte I made the necessary changes and the issue is finally resolved Thank you @kgryte .
You can review it.

**Review by Planeshifter — APPROVED** [review]
Thanks, @vr-varad! Looked all good now; will merge shortly.

**Pull request merged**